storage/kvstore/filesystemstore/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+target_include_directories(mbed-storage-kvstore
```

Comment:
This should be componentised into `mbed-storage-tdbstore` or `mbed-storage-kvstore-tdbstore`. It's outside the scope of this task, so I'm leaving componentisation to the CMake team.

Reply:
we would like to work with you to fix this, can you please create a story in your board to do this, we will assist.

Examples show how this is not ideally structured (for kv store I need to add like 5 libraries), we should fix dependencies.

Reply:
What structure would you proposed?

For BlockDevice, we have one CMake target per type of BlockDevice: https://github.com/ARMmbed/mbed-os/blob/66c05ddbf8da7873108494ec9d243c2d2ca6a283/storage/CMakeLists.txt#L7-L13

And in `mbed-storage-blockdevice`, we include them based on availability: https://github.com/ARMmbed/mbed-os/blob/66c05ddbf8da7873108494ec9d243c2d2ca6a283/storage/blockdevice/CMakeLists.txt#L4-L26

Shall we go for the same approach for KVStore?

Reply:
@LDong-Arm ,
One of the reasons for merging the CMake work to master was so it becomes everyone's responsibility to update CMake input source files according to changes made in the file structure.
We can help you review the CMake input source files but you have to make the necessary change along with everything else.
The approach used for BlockDevice can also be used here.
Thanks.

Reply:
The PR in the current state already works as it is, aligning with the BlockDevice approach is a further suggestion. I can still make it here in this PR, since it's a simple change.

Reply:
Your PR slims down what is included for baremetal.|
This is currently not the case if the `mbed-baremetal` CMake target is used with `mbed-storage-kvstore`.

Reply:
Done, please review again, thanks